src/istgt_iscsi.c

```diff
@@ -2049,6 +2053,22 @@ istgt_iscsi_op_login(CONN_Ptr conn, ISCSI_PDU_Ptr pdu)
 			lu = NULL;
 			tsih = 0;
 
+			__sync_add_and_fetch(&discovery_counter, 1);
+
+			/*
+			 * As per the issue https://github.com/openebs/openebs/issues/2382,
+			 * 5 continous discovery requests failure can cause iscsi initiator
+			 * to get into high CPU usage.
+			 * This code checks for this state of 4 continuous discovery failures
+			 * and restarts the process.
+			 */
+			if (discovery_counter > 4) {
```

Comment:
You might want to make this value configurable

Reply:
configurable make sense. But, it involves lot of code change for this throw away code.
An environmental variable can be added, if that makes sense. Should I go ahead with an environmental variable?

Reply:
Yeah, that might be a good idea. getenv() could be used, and default to 4 if not found. We should also make sure a log entry exists about the currently used setting.